### PR TITLE
json writer, refact switch `case`s with `return` to gain a few sloc

### PR DESCRIPTION
**Pull request by ghost**
I discover this semantic oddity only today... don't you as well ?

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by Basile-z** [comment]
looking at the Azure Pipeline log...maybe this semantic oddity is a bug actually (accept invalid).

File: src/dmd/json.d [hunk 1]
```diff
@@ -93,27 +93,13 @@ public:
...
-            case '\\':
-                buf.writestring("\\\\");
-                break;
-            case '\b':
-                buf.writestring("\\b");
-                break;
-            case '\f':
-                buf.writestring("\\f");
-                break;
+            case '\n': return buf.writestring("\\n");
```
Comment (Basile-z) [inline src/dmd/json.d]: this one is in a loop, cant be modified

**Comment by thewilsonator** [comment]
```
    extern(D) void stringPart(const char[] s)
    {
        foreach (char c; s)
	{
	            switch (c)
                        // do some returns
       }
}
```

File: src/dmd/json.d [hunk 2]
```diff
@@ -94,7 +94,7 @@ public:
             switch (c)
             {
             case '\n':
-                buf.writestring("\\n");
+                return buf.writestring("\\n");
```
Comment (thewilsonator) [inline src/dmd/json.d]: missed one

**Comment by Basile-z** [comment]
yeah tanks @thewilsonator , just noticed too.

**Pull request merged**